### ISSUE TITLE
Crash in <<~ squiggly heredoc syntax

### ISSUE THREAD

**Issue by bessey** [ruby-formatter/rufo#6]
Reproduce with:
```rb
#ERRORS
      <<~HTML
          There are #{blah} #{bloop}
      HTML
```

Replace `~` with `-` and the formatter works fine

```rb
#OK
      <<~HTML
          There are #{blah} #{bloop}
      HTML
```

Delete one interpolation and the formatter works fine.

Interestingly, and almost certainly related, 
```rb
<<-TEXT
  There are    #{blah}    thing
TEXT
```
gets formatted succesfully, but rufo wrongly removes whitespace inside the multiline string:
```rb
# space between #{blah} and thing reduced
<<~TEXT
  There are    #{blah}  thing 
TEXT
```

**Comment by bessey** [comment]
Minimum case appears to be:
- a squiggly doc string
- with 2+ interpolations
- with at least one space before the last interpolation

```rb
assert_format "<<~EOF\n\#{1}\#{2}\nEOF" # passes 
assert_format "<<~EOF\n \#{1}\#{2}\nEOF" # errors
assert_format "<<~EOF\n\#{1} \#{2}\nEOF" # errors
assert_format "<<~EOF\n\#{1} \#{2} \nEOF"
# fails, because rufo strips the end space, which I'm pretty sure is wrong?
# shouldn't we never strip trailing spaces in a string?
```

**Comment by bessey** [comment]
Well I had a mess around but got nowhere beyond reproducing with a minimal test case. My work is on [tilde-heredoc-interpolation-bug](https://github.com/ruby-formatter/rufo/tree/tilde-heredoc-interpolation-bug) if anyone wants to play around.

**Comment by mjago** [comment]
Errors:
1. `assert_format "<<~EOF\n \#{1}\#{2}\nEOF"` >>
`Expected token on_int, not on_tstring_content at [[2, 1], :on_tstring_content, ""]`

2. `assert_format "<<~EOF\n\#{1} \#{2}\nEOF"`
`assert_format "<<~EOF\n\#{1} \#{2} \nEOF"` >>
`Expected token on_tstring_content, not on_embexpr_beg at [[2, 5], :on_embexpr_beg, "\#{"]`

**Comment by mjago** [comment]
I fixed upto head but with head there is another issue which looks like another ripper bug:

``` 
ruby -rripper -rpp -e 'pp Ripper.lex("<<~ABC\n \#{1}   \#{2}\nABC")   '
[[[1, 0], :on_heredoc_beg, "<<~ABC"],
 [[1, 6], :on_nl, "\n"],
 [[2, 0], :on_ignored_sp, " "],
 [[2, 1], :on_embexpr_beg, "\#{"],
 [[2, 3], :on_int, "1"],
 [[2, 4], :on_embexpr_end, "}"],
 [[2, 5], :on_ignored_sp, " "],
 [[2, 6], :on_tstring_content, "  "],
 [[2, 8], :on_embexpr_beg, "\#{"],
 [[2, 10], :on_int, "2"],
 [[2, 11], :on_embexpr_end, "}"],
 [[2, 12], :on_tstring_content, "\n"],
 [[3, 0], :on_heredoc_end, "ABC"]]```

Ripper sees the space between embedded expressions as ignored space as it does the space following the newline (which is ignored_ space). I'll raise an issue for it.

**Comment by mjago** [comment]
Head is fixed, but prior to 2.5 looks unfixable for squiggly heredocs having space between embedded expressions since the whitespace information is lost by Ripper (see issue https://bugs.ruby-lang.org/issues/13536). The crash is however fixed with #33.

**Comment by bessey** [comment]
Is the ignored space thing actually a bug? Isn't that Ripper acknowledging that in squiggly syntax these are equivalent, because of the built in heredoc stripping:
```ruby
<<~ABC
  text text
    text
ABC

<<~ABC
            text text
              text
ABC

**Comment by mjago** [comment]
Sure the leading space is ignored but not the space between the embedded expressions.

**Comment by mjago** [comment]
Fixed with  8bab7a2  #33
